api/pkg/metadata/service.go

```diff
@@ -68,3 +74,47 @@ func (s *APIService) ListMetadata(request *restful.Request, response *restful.Re
 	log.Info("Get metadata details successfully.")
 	response.WriteEntity(res.Buckets)
 }
+
+//* This function fetches the request parameters from the request and assigns them default values if not present.
+//* It returns ListMetadataRequest for ListMetaData API call
+func GetListMetaDataRequest(request *restful.Request) (listMetadataRequest mt.ListMetadataRequest, err error) {
+	typeOfCloudVendor := request.PathParameter("type")
+	limitStr := request.PathParameter("limit")
+	backendName := request.PathParameter("backendName")
+	offsetStr := request.PathParameter("offset")
+	bucketName := request.PathParameter("bucketName")
+	objectName := request.PathParameter("objectName")
+	sizeOfObject := request.PathParameter("sizeOfObject")
+	sizeOfBucket := request.PathParameter("sizeOfBucket")
+
+	offsetStr, limitStr = assignDefaultValues(offsetStr, limitStr)
+
+	// convert string offset to int32
+	offset, err := strconv.ParseInt(offsetStr, 10, 32)
```

Comment:
You may use  GetPaginationParam() function from common.go

Reply:
done